src/ser.rs

```diff
@@ -65,8 +65,8 @@ impl<'a, W: enc::Write> serde::Serializer for &'a mut Serializer<W> {
     type SerializeTupleStruct = BoundedCollect<'a, W>;
     type SerializeTupleVariant = BoundedCollect<'a, W>;
     type SerializeMap = CollectMap<'a, W>;
-    type SerializeStruct = BoundedCollect<'a, W>;
-    type SerializeStructVariant = BoundedCollect<'a, W>;
+    type SerializeStruct = CollectMap<'a, W>;
```

Comment:
Can you educate me on BoundedCollet vs CollectMap?

Reply:
The names might not be great, but the idea now is that `BoundedCollect` does take the input in the same order as it comes in without further checks, the `CollectMap` does the whole sorting thing.

Those structs then implement specific traits so that they can be used with those type definitions.